rpc/subscriptions_test.go

```diff
@@ -326,7 +330,533 @@ func TestSubscribeEvents(t *testing.T) {
 	})
 }
 
-func marshalSubscriptionResponse(e *EmittedEvent, id uint64) ([]byte, error) {
+type fakeSyncer struct {
+	newHeads   *feed.Feed[*core.Header]
+	reorgs     *feed.Feed[*sync.ReorgBlockRange]
+	pendingTxs *feed.Feed[[]core.Transaction]
+}
```

Comment:
I am a little confused why we have this? Is mockSyncReader not sufficient?

Reply:
Please ignore this, I realised the MockSyncReader isn't enough to send values through the channel.